torch/csrc/jit/codegen/cuda/shape_inference.cpp

```diff
@@ -492,18 +483,43 @@ class NaiveTypePropagator {
           ? *scalar_type
           : c10::promoteTypes(*op0->scalarType(), *op1->scalarType());
 
-      return TensorType::create(
-          promoted_scalar_type, *op0->device(), c10::nullopt, c10::nullopt);
+      if (op0->isComplete()) {
+        return TensorType::create(
+            promoted_scalar_type,
+            *op0->device(),
+            op0->sizes(),
+            op0->strides(),
+            c10::nullopt);
+      } else if (op1->isComplete()) {
+        return TensorType::create(
+            promoted_scalar_type,
+            *op1->device(),
+            op1->sizes(),
+            op1->strides(),
+            c10::nullopt);
+      } else {
+        return TensorType::create(
+            promoted_scalar_type, *op0->device(), c10::nullopt, c10::nullopt);
+      }
     } else {
       auto ptr = (op0 != nullptr) ? op0 : op1;
       TORCH_CHECK(
           hasTypeAndDevice(ptr),
           "Type and device propagation has failed, or was not provided enough information.");
-      return TensorType::create(
-          scalar_type.has_value() ? *scalar_type : *ptr->scalarType(),
-          *ptr->device(),
-          c10::nullopt,
-          c10::nullopt);
+      if (ptr->isComplete()) {
```

Comment:
ditto

Reply:
Isn't one of the arguments a scalar value in this case? So, why can't we take the shape of the tensor if it available?

Reply:
My bad, in case of scalar value, yes we should just be able to forward tensor properties.